File: backend/user-service/src/main/java/com/skylink/user/service/UserService.java
package com.skylink.user.service;

import com.skylink.user.dto.request.UpdateProfileRequest;
import com.skylink.user.dto.response.ApiResponse;
import com.skylink.user.dto.response.UserProfileResponse;

import java.util.List;

public interface UserService {

    ApiResponse<UserProfileResponse> getProfile(Long id);

    ApiResponse<UserProfileResponse> updateProfile(
            Long id,
            UpdateProfileRequest request
    );

    ApiResponse<List<UserProfileResponse>> getAllUsers();

    ApiResponse<String> deactivateUser(Long id);
}

File: frontend/src/api/userService.ts
import axiosClient from './axiosClient';

export const userService = {
  createProfile: async (userData) => {
    const response = await axiosClient.post('/users', userData);
    return response.data;
  },
  getProfile: async (id) => {
    const response = await axiosClient.get(`/users/${id}`);
    return response.data;
  },
  updateProfile: async (id, userData) => {
    const response = await axiosClient.put(`/users/${id}`, userData);
    return response.data;
  }
};
